src/peak_finding/1d_list.clj

```diff
@@ -0,0 +1,54 @@
+;; A peak is any element that is larger or equal to both it's neighbours
+;; Items that only have one neighbour (end items) are considered a peak if
+;; they are larger or equal to their neighbour
+
+(ns peakfinder
+  (:require [clojure.core.match :refer [match]]))
+
+(defun find-peak [input]
+  "Returns all peaks in a sequence"
+  (match [input]
+         ;; checks input is a sequence of multiple numbers
+         [(input :guard #(and (sequential? %)
```

Comment:
Instead of checking for (> (count %) > 1) you could just put the [([h] :seq)] case ahead of this case.